singularity-eos/base/table_bounds.hpp

```diff
@@ -0,0 +1,281 @@
+//------------------------------------------------------------------------------
+// © 2021-2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef SINGULARITY_EOS_BASE_TABLE_BOUNDS_HPP_
+#define SINGULARITY_EOS_BASE_TABLE_BOUNDS_HPP_
+#ifdef SINGULARITY_USE_SPINER
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include <ports-of-call/portability.hpp>
+#include <ports-of-call/portable_errors.hpp>
+#include <singularity-eos/base/fast-math/logs.hpp>
+#include <spiner/databox.hpp>
+#include <spiner/interpolation.hpp>
+
+namespace singularity {
+
+// For logarithmic interpolation, quantities may be negative.
+// If they are, use offset to ensure negative values make sense.
+template <int NGRIDS = 3>
+class Bounds {
+ public:
+  using RegularGrid1D = Spiner::RegularGrid1D<Real>;
+  using Grid_t = Spiner::PiecewiseGrid1D<Real, NGRIDS>;
+  // for tag dispatch in constructors
+  using OneGrid = std::integral_constant<int, 1>;
+  using TwoGrids = std::integral_constant<int, 2>;
+  using ThreeGrids = std::integral_constant<int, 3>;
+
+  Bounds() : offset(0), piecewise(false), linmin_(0), linmax_(0) {}
+
+  Bounds(Real min, Real max, int N, Real offset)
+      : grid(Grid_t(std::vector<RegularGrid1D>{RegularGrid1D(min, max, N)})),
+        offset(offset), piecewise(false), linmin_(min), linmax_(max) {}
+  Bounds(OneGrid, Real min, Real max, int N, Real offset) : Bounds(min, max, N, offset) {}
+
+  Bounds(Real min, Real max, int N, bool convertToLog = false, Real shrinkRange = 0,
+         Real anchor_point = std::numeric_limits<Real>::signaling_NaN())
+      : offset(0), piecewise(true), linmin_(min), linmax_(max) {
```

Comment:
It looks like several of these arguments can be `const`.

Rather than commenting on each constructor, can you make the other constructors have `const` arguments where appropriate?

Reply:
Most of these arguments **cannot** be `const`. However, I think that is a strong reason to name the ones that are explicitly const.